### PR TITLE
Update to worth with numpy >= 1.24

### LINKED ISSUE #19: behave_pandas library does not work for numpy >= 1.24
- As from numpy 1.24 [(release notes)](https://numpy.org/doc/stable/release/1.24.0-notes.html#expired-deprecations) numpy.float, numpy.int and similar aliases were removed from the library , causing errors when they are used.

- Since there are occurrences of np.float and np.int in the code,  they should be replaced with plain Python float and int.

Source : https://stackoverflow.com/questions/74844262/how-to-solve-error-numpy-has-no-attribute-float-in-python

### PR DESCRIPTION
**Pull request by lkacenja**
We use this great tool as part of our automated testing here at the foundation. Thanks for all the great work! This PR removes usage of the deprecated numpy base types (np.float, np.int, np.bool). It also updates a few of the test steps to work with some more recent changes to pandas.

closes #19

### CODE REVIEW

**Comment by clembou** [comment]
Thanks @lkacenja ! this is all looking good but while reviewing this I noticed the CI automation broke on this repo. 
I switched to Github Actions and got it going again in: https://github.com/clembou/behave-pandas/issues/19

Note that of course to get a green build I had to pin `pandas` and `numpy` in `setup.py`:
```python
install_requires=["behave", "pandas>=0.21,<1.5", "tabulate", "numpy<1.24"],
```
I have to keep this library working with python 3.6 ideally (still used at work in a few places), can you please merge or rebase this PR on top of master, and see what happens on older versions with this change?
you'll also have to revert the `setup.py` line to:
```python
install_requires=["behave", "pandas>=0.21", "tabulate"],
```
to ensure newer python versions do install the newly supported `pandas` and `numpy` versions. (python 3.6 will install pandas 1.1.5 automatically as this is the last supported version on 3.6 - this is what I need to test).

If the build goes red, and supporting both legacy versions and the latest is hard I can drop 3.6 form the support matrix, and work around this by pinning the dependency in my work repos where relevant, but with luck the new syntax can work even on older `pandas` and `numpy` - let's see!

File: features/printer.feature [hunk 1]
```diff
@@ -5,35 +5,35 @@ Feature: Table printer
 
   Scenario: simple index and valid data
     Given a gherkin table as input
-      | str       | float     | str                 | dict      | list     | OrderedDict              | OrderedDict      |
-      | index_col | float_col | str_col             | dict_col  | list_col | ordered_dict_col         | ordered_dict_col |
-      | egg       | 3.0       | silly walks         | {}        | []       | OrderedDict()            | []               |
-      | spam      | 4.1       | spanish inquisition | {"a": 1}  | ["a"]    | OrderedDict([("a", 1)])  | [("a", 1)]       |
-      | bacon     | 5.2       | dead parrot         | {"a": []} | [1]      | OrderedDict([("a", [])]) | [("a", [])]      |
+      | str       | float     | str                 | dict      | list     | OrderedDict              | OrderedDict        |
+      | index_col | float_col | str_col             | dict_col  | list_col | ordered_dict_col         | ordered_dict_col_2 |
```
Comment (clembou) [inline features/printer.feature]: I already pulled this change in master, I tried to do the exact same change as you so hopefully you won't get conflicts

**Comment by lkacenja** [comment]
We use Gitlab for most of our work, so I'm less familiar with Github CI. I rebased this master branch into the forked master branch. I _think_ the CI is running green [here](https://github.com/morrisanimalfoundation/behave-pandas/actions). Let me know if I'm missing anything. Thanks!

**Review by clembou — APPROVED** [review]
Tested locally the tests pass. Github CI also seem ok with the PR. We'll need to remove the upper bound restrictions in the `setup.py` file before cutting a release, I'll take care of this. Thanks for your contribution!

**Comment by greggles** [comment]
@clembou Thanks for your help in maintaining this library and the quick feedback/reviews on this merge review.

**Pull request merged**